### ISSUE TITLE
RSS feed shows draft posts and displays in an unexpected order (proposed fix included)

### ISSUE THREAD

**Issue by davidvkimball** [saicaca/fuwari#118]
Here's how I modified it to behave the way I expected: 

```
import rss from '@astrojs/rss';
import { siteConfig } from '@/config';
import { getCollection } from 'astro:content';
import sanitizeHtml from 'sanitize-html';
import MarkdownIt from 'markdown-it';
const parser = new MarkdownIt();

export async function GET(context: any) {
  const blog = await getCollection('posts');
  return rss({
    title: siteConfig.title,
    description: siteConfig.subtitle || 'No description',
    site: context.site,
    items: blog
      .filter(post => !post.data.draft)  // Exclude drafts
      .sort((a, b) => new Date(b.data.published).getTime() - new Date(a.data.published).getTime())  // Sort by published date
      .map(post => ({
        title: post.data.title,
        pubDate: post.data.published,
        description: post.data.description,
        link: `/posts/${post.slug}/`,
        content: sanitizeHtml(parser.render(post.body), {
          allowedTags: sanitizeHtml.defaults.allowedTags.concat(['img'])
        }),
      })),
    customData: `<language>${siteConfig.lang}</language>`,
  });
}
```

Instead of using `getCollection('posts')`, I just used the publish date as defined in the frontmatter.

Also, for anyone using satori, I used this to add OpenGraph images to my RSS feed, so services like MailChimp can import them. 

```
import rss from '@astrojs/rss';
import { siteConfig } from '@/config';
import { getCollection } from 'astro:content';
import sanitizeHtml from 'sanitize-html';
import MarkdownIt from 'markdown-it';
const parser = new MarkdownIt();

export async function GET(context: any) {
  const blog = await getCollection('posts');
  return rss({
    title: siteConfig.title,
    description: siteConfig.subtitle || 'No description',
    site: context.site,
    xmlns: {
      media: "http://search.yahoo.com/mrss/",
    },
    items: blog
      .filter(post => !post.data.draft)  // Exclude drafts
      .sort((a, b) => new Date(b.data.published).getTime() - new Date(a.data.published).getTime())  // Sort by published date
      .map(post => {
        return {
          title: post.data.title,
          pubDate: post.data.published,
          description: post.data.description,
          link: `${context.site}posts/${post.slug}/`,
          content: sanitizeHtml(parser.render(post.body), {
            allowedTags: sanitizeHtml.defaults.allowedTags.concat(['img'])
          }),
         // Add Open Graph images
          customData: `<media:content
              type="image/png"
              width="564px"
              height="296px"
              url="${context.site}open-graph/${post.slug}.png" />
          `,
        };
      }),
    customData: `<language>${siteConfig.lang}</language>`,
  });
}
```

Hope this helps!

**Comment by saicaca** [comment]
Fixed in 2622aabe075d836868971e0f9abe73267213cc78. Thank you for spotting the bug!

**Comment by davidvkimball** [comment]
You're welcome! @saicaca is it possible to be listed as a contributor on this project due to my contribution here? Or did you end up using a different fix?
